### PR TITLE
[new TestSuite] add OSUTestSuite

### PR DESCRIPTION
**Pull request by darkwhite29**
Added OSU Micro-Benchmarks as a new test.

### CODE REVIEW

File: microsoft/runbook/azure/alma8_gpu_driver.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: azure default
+include:
```
Comment (dsrivastavv) [inline microsoft/runbook/azure/alma8_gpu_driver.yml]: We can have a common runbook and pass parameters at runtime. For example: 
```
./lisa.sh -v "resource_group_name:<resource_group_name>"

This will ensure that there are few runbooks which are also easier to maintain. Further, we can also use combinator where you can specify these images and they will be picked at runtime. More info on combinator can be found [here](https://mslisa.readthedocs.io/en/main/write_test/extension.html#combinator)

File: microsoft/testsuites/gpu/gpusuite.py [hunk 2]
```diff
@@ -16,10 +17,12 @@
     TestSuiteMetadata,
     constants,
     simple_requirement,
+    CustomScript,
+    CustomScriptBuilder
 )
 from lisa.features import Gpu, GpuEnabled, SerialConsole
 from lisa.features.gpu import ComputeSDK
-from lisa.operating_system import AlmaLinux, Debian, Oracle, Suse
```
Comment (dsrivastavv) [inline microsoft/testsuites/gpu/gpusuite.py]: We don't need to remove these. In you tests, you can simply add a check to run test if the images is Debian

File: microsoft/testsuites/gpu/gpusuite.py [hunk 3]
```diff
@@ -100,7 +109,7 @@ def verify_gpu_adapter_count(self, node: Node, log_path: Path, log: Logger) -> N
             "Expected device count didn't match Actual device count from lspci",
         ).is_equal_to(expected_count)
 
-        _check_driver_installed(node)
+        ##_check_driver_installed(node)
```
Comment (dsrivastavv) [inline microsoft/testsuites/gpu/gpusuite.py]: Can you also please cleanup the comments?

File: microsoft/testsuites/gpu/gpusuite.py [hunk 4]
```diff
@@ -191,23 +200,20 @@ def verify_gpu_cuda_with_pytorch(self, node: Node) -> None:
...
-        raise SkippedException(
-            f"{node.os.name} doesn't support GPU driver installation."
-        )
-
     try:
         _ = node.tools[NvidiaSmi]
     except Exception as identifier:
         raise LisaException(
-            f"Cannot find nvidia-smi, make sure the driver installed correctly. "
+            f"Cannot find nvidia-smi {node.os.name}, make sure the driver installed correctly. "
```
Comment (dsrivastavv) [inline microsoft/testsuites/gpu/gpusuite.py]: os name info will be available in the logs. We avoid redundancy in code and avoid logging values which can be inferred from the logs

File: microsoft/testsuites/gpu/gpusuite.py [hunk 5]
```diff
@@ -245,13 +251,29 @@ def _ensure_driver_installed(
...
+    '''node.execute(
+        ##f"./microsoft/runbook/azure/install_dependencies_alma.sh alma8.6",
+        ##f"touch /tmp/test29",
+        f"./test.sh",
+        shell=True,
+        sudo=True,
+    )'''
+    ##node.execute("./test.sh, cwd=./microsoft/testsuites/gpu, shell=True, sudo=True")
+
+    test_script = CustomScriptBuilder(Path(__file__).parent.joinpath("."), ["install_dependencies_alma.sh"])
```
Comment (dsrivastavv) [inline microsoft/testsuites/gpu/gpusuite.py]: This should be seperated into multiple tools. The scripts are hard to maintain. Please let us know if you are finding it difficult to implement any tools. Further, you can use `node.execute` to run a command directly

File: microsoft/testsuites/xfstests/xfstests.py [hunk 6]
```diff
@@ -9,447 +9,448 @@
 from lisa.executable import Tool
 from lisa.messages import SubTestMessage, TestStatus, create_test_result_message
 from lisa.operating_system import (
-    CBLMariner,
-    Debian,
-    Oracle,
```
Comment (dsrivastavv) [inline microsoft/testsuites/xfstests/xfstests.py]: Can you clarify why changes were intended in this file?

File: microsoft/testsuites/osu/osusuite.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+from lisa.features import Gpu, GpuEnabled, SerialConsole
+from lisa.features.gpu import ComputeSDK
+from lisa.operating_system import Debian
+from lisa.tools import Lspci, NvidiaSmi, Pip, Python, Reboot, Service, Tar, Wget
+from lisa.util import get_matched_str
+
+_osu_mpi_location = ("https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-2.3.7-1.tar.gz")
+_osu_mpi_file_name = "mvapich2.tgz"
+
+_osu_bench_location = ("http://mvapich.cse.ohio-state.edu/download/mvapich/osu-micro-benchmarks-6.2.tar.gz")
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: Global variables are capital snake case. More guidelines can be found here: https://mslisa.readthedocs.io/en/main/write_test/guidelines.html

File: microsoft/testsuites/osu/osusuite.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        """,
+        timeout=TIMEOUT,
+        requirement=simple_requirement(
+            supported_features=[GpuEnabled(), SerialConsole],
+        ),
+        priority=1,
+    )
+    def run_osu_bench(self, node: Node, log_path: Path, log: Logger) -> None:
+
+        _install_osu_mpi(node)
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py:60]: These should be private functions in class

File: microsoft/testsuites/osu/osusuite.py [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+        gpu_count = node.capability.gpu_count
+        gpu = node.features[Gpu]
+        expected_count = gpu.get_gpu_count_with_lspci()
+        assert_that(gpu_count).described_as(
+            "Expected device count didn't match actual device count from lspci!"
+        ).is_equal_to(expected_count)
+
+        '''test_script = CustomScriptBuilder(Path(__file__).parent.joinpath("."), ["install_dependencies_alma.sh"])
+        script: CustomScript = node.tools[test_script]
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: Can you please reframe this in simple words? We avoid putting code in comments and rather describe the logic in simple words

File: microsoft/testsuites/osu/osusuite.py [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+        result = script.run("alma8.6")'''
+        node.execute(
+            f"mpirun -np {gpu_count} python3 {path}/python/run.py --benchmark allgather --min 128 --max 10000000 >> /tmp/osu_bench.log", #--buffer pycuda #--buffer cupy
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: Appending to a log file will result in progress not being displayed on the lisa logs. Is this file being used somewhere? Also, please make a tools for `mpirun`

File: microsoft/runbook/azure/alma8_gpu_driver.yml (hunk 1 above)
Comment (darkwhite29) [reply microsoft/runbook/azure/alma8_gpu_driver.yml]: I would like to keep the current multiple runbook yaml files, which gives me an idea of which images and tests we have covered.

File: microsoft/testsuites/gpu/gpusuite.py (hunk 2 above)
Comment (darkwhite29) [reply microsoft/testsuites/gpu/gpusuite.py]: I think I mistakenly removed this line. Will be reverted.

File: microsoft/testsuites/gpu/gpusuite.py (hunk 3 above)
Comment (darkwhite29) [reply microsoft/testsuites/gpu/gpusuite.py]: Done.

File: microsoft/testsuites/gpu/gpusuite.py (hunk 4 above)
Comment (darkwhite29) [reply microsoft/testsuites/gpu/gpusuite.py]: I added it for debugging purposes. Forgot to remove. Just done.

File: microsoft/testsuites/gpu/gpusuite.py (hunk 5 above)
Comment (darkwhite29) [reply microsoft/testsuites/gpu/gpusuite.py]: I was told by Chi that I should use CustomScriptBuilder to upload scripts to remove node to run. `node.execute` will only run local scripts. In this case it does not work.

Comment (darkwhite29) [reply microsoft/testsuites/gpu/gpusuite.py]: I left the `node.execute` block (commented out) just for reminder. Please let me know if you want me to remove it.

File: microsoft/testsuites/xfstests/xfstests.py (hunk 6 above)
Comment (darkwhite29) [reply microsoft/testsuites/xfstests/xfstests.py]: I am just copying a recent change by Lili. This is not my file. Reverted.

File: microsoft/testsuites/osu/osusuite.py (hunk 7 above)
Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: Understandable, but I am just using the format in this test: https://github.com/microsoft/lisa/blob/ee11364108da6c6e68cef988d11c9301d41e55ed/microsoft/testsuites/gpu/gpusuite.py

File: microsoft/testsuites/osu/osusuite.py (hunk 8 above)
Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py:60]: Fixed.

File: microsoft/testsuites/osu/osusuite.py (hunk 9 above)
Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: Removed.

File: microsoft/testsuites/osu/osusuite.py (hunk 10 above)
Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: I used this log file for debugging purposes. Removed.

File: microsoft/testsuites/osu/osusuite.py (hunk 7 above)
Comment (dsrivastavv) [reply microsoft/testsuites/osu/osusuite.py]: That should be changed as well. This is probably a old code which skipped these quality checks. Can you please correct it from your end? I will make a note to correct any remaining instances myself

File: microsoft/testsuites/xfstests/xfstests.py (hunk 6 above)
Comment (dsrivastavv) [reply microsoft/testsuites/xfstests/xfstests.py]: Can you please rebase from the latest main branch instead?

File: microsoft/testsuites/gpu/gpusuite.py (hunk 5 above)
Comment (dsrivastavv) [reply microsoft/testsuites/gpu/gpusuite.py]: Can you please remove these comments before we merge the PR? We want to make sure that only clean changes are going to the main branch

File: microsoft/runbook/azure/alma8_gpu_driver.yml (hunk 1 above)
Comment (dsrivastavv) [reply microsoft/runbook/azure/alma8_gpu_driver.yml]: I think using combinator would be much better. i can give you an example. Can you please tell what variables change in each runbook?

File: microsoft/testsuites/osu/osusuite.py [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+def _install_osu_bench(node: Node) -> str:
+
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = wget.get_tool_path(use_global=True)
+    extracted_path = tar.get_tool_path(use_global=True)
+    if node.shell.exists(path / OSU_BENCH_FILE_NAME):
+        return
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: Is this expected return value? The return type of this function is `str`

Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: Yes, I was trying to return a file path.

Comment (dsrivastavv) [reply microsoft/testsuites/osu/osusuite.py]: The `_install_osu_bench ` is finishing the install and returning the location. The returned location is used in Ln 67. This line is however returning none, which will lead to failure in Ln 67. Can you please instead check if `osu-micro-benchmarks` executable is present and return it's location similar to to Ln 128?

File: microsoft/testsuites/osu/osusuite.py [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+    extracted_path = tar.get_tool_path(use_global=True)
+    if node.shell.exists(path / OSU_MPI_FILE_NAME):
+        return
+
+    download_path = wget.get(
+        url=OSU_MPI_LOCATION, filename=str(OSU_MPI_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(extracted_path))
+
+    target_path = "/usr/local/"
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: [nit] This doesn't need to be a variable since it's used just once. You can use the value directly

File: microsoft/testsuites/osu/osusuite.py [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+
+    return
+
+def _install_osu_bench(node: Node) -> str:
+
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = wget.get_tool_path(use_global=True)
+    extracted_path = tar.get_tool_path(use_global=True)
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: `get_tool_path` will give you location where the the tool is located which is one of the system folder. We don't need to install or extract files here. You can use `lisa_working_dir` for doing extracting files and compiling any program. This can be accessed with `node.get_working_path`.

File: microsoft/testsuites/osu/osusuite.py [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+    if node.shell.exists(path / OSU_BENCH_FILE_NAME):
+        return
+
+    download_path = wget.get(
+        url=OSU_BENCH_LOCATION, filename=str(OSU_BENCH_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(extracted_path))
+
+    node.execute("apt-get install -y python3.8-dev", shell=True, sudo=True)
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: Please use `node.os.install_packages`

File: microsoft/testsuites/osu/osusuite.py [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+    node.execute("pip3 install numpy", shell=True, sudo=True)
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: All the pip install can be combined into a single command. Something like `node.execute("pip3 install numpy numba ...)"` should work.

File: microsoft/testsuites/osu/osusuite.py [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+    if node.shell.exists(path / OSU_MPI_FILE_NAME):
+        return
+
+    download_path = wget.get(
+        url=OSU_MPI_LOCATION, filename=str(OSU_MPI_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(extracted_path))
+
+    target_path = "/usr/local/"
+    node.execute(
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py:119]: You don't need to use `cd`. You can use `cwd` option in the node.execute. This will run the command from the given folder. For example:
```
node.execute("./configure --prefix {target_path}", shell=True, cwd = f"{extracted_path}/mvapich2-2.3.7-1")
```

Further, sudo is false by default. So we don't need to specify default arguments in the function call

File: microsoft/testsuites/osu/osusuite.py [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+        f"cd {extracted_path}/mvapich2-2.3.7-1; ./configure --prefix {target_path}",
+        shell=True,
+        sudo=False,
+    )
+    node.execute(f"cd {extracted_path}/mvapich2-2.3.7-1; make", shell=True, sudo=False)
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: Can you please use the `make` tool with `cwd` option? It has both `make` and `make_install` functions present

File: microsoft/testsuites/osu/osusuite.py [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+    node.execute(
+        f"cd {extracted_path}/mvapich2-2.3.7-1; make install", shell=True, sudo=True
+    )
+
+    return
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: [nit] this is not required since this will return None if no return is specified

**Comment by darkwhite29** [comment]
I cannot see your comments on GitHub, if you are referring to
Divyansh's suggestion to use node.os.install_packages to install
package, node.os.install_packages gives me some errors when checked
using nox -vRs mypy. As such, I kept the old way to install package.

Thanks.

--
Li

File: microsoft/testsuites/osu/osusuite.py (hunk 13 above)
Comment (dsrivastavv) [reply microsoft/testsuites/osu/osusuite.py]: I think Lili's comment was that this is not entirely fixed. Can you please try removing `get_tool_path` from everywhere and work with respect to working_dir?

**Comment by dsrivastavv** [comment]
Can you please paste the error?

File: microsoft/testsuites/osu/osusuite.py (hunk 13 above)
Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: Currently,
`path = wget.get_tool_path(use_global=True)`
`extracted_path = node.get_working_path()`
where path is the path to download packages, and extracted_path is the path to extract them. So you want me to use the same path to download and extract?

**Comment by darkwhite29** [comment]
> Can you please paste the error?

I used up all my codespace and cannot retrieve the complete error now. According to my memory, it says: `node.os` does not have function `install_packages()`.

I checked all the existing LISA tests and found they are using `self.node.os.install_packages`, but when I put `self` in the front, there were new errors coming. So eventually I stick to use `node.execute()`

**Comment by dsrivastavv** [comment]
> > Can you please paste the error?
> 
> I used up all my codespace and cannot retrieve the complete error now. According to my memory, it says: `node.os` does not have function `install_packages()`.
> 
> I checked all the existing LISA tests and found they are using `this.node.os.install_packages`, but when I put `this` in the front, there were new errors coming. So eventually I stick to use `node.execute()`

Can you please try this:
```
 assert isinstance(
            node.os, Posix
        ), f"unzip: unsupported OS {self.node.os.name}"
 node.os.install_packages(<package-name>)
 ```

**Comment by darkwhite29** [comment]
> > > Can you please paste the error?
> > 
> > 
> > I used up all my codespace and cannot retrieve the complete error now. According to my memory, it says: `node.os` does not have function `install_packages()`.
> > I checked all the existing LISA tests and found they are using `this.node.os.install_packages`, but when I put `this` in the front, there were new errors coming. So eventually I stick to use `node.execute()`
> 
> Can you please try this:
> 
> ```
>  assert isinstance(
>             node.os, Posix
>         ), f"unzip: unsupported OS {self.node.os.name}"
>  node.os.install_packages(<package-name>)
> ```

Updated and all nox tests passed.

File: microsoft/testsuites/osu/osusuite.py (hunk 13 above)
Comment (dsrivastavv) [reply microsoft/testsuites/osu/osusuite.py]: We can do something like this:

```
_download_path = node.get_working_path()
download_path = wget.get(
        url=OSU_MPI_LOCATION, filename=str(OSU_MPI_FILE_NAME), file_path=str(_download_path)
    )
tar.extract(download_path, dest_dir=str(_download_path / <compile folder name>))
```
There are two instance of `get_tool_path` that need to be changed

Comment (dsrivastavv) [reply microsoft/testsuites/osu/osusuite.py]: You can also try `tar.extract(download_path, dest_dir=str(_download_path))` directly. I will leave it to you on how you want to structure installs in the `get_working_path` directory

Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: Updated and nox tests passed. Please take a final look. Thanks.

**Review by dsrivastavv — DISMISSED** [review]
LGTM! @squirrelsc / @LiliDeng Can you please take a look?

@darkwhite29 Please fix any pipeline issues

**Comment by darkwhite29** [comment]
> LGTM! @squirrelsc / @LiliDeng Can you please take a look?
> 
> @darkwhite29 Please fix any pipeline issues

You mean all issues from "nox -vRt all"? There are no issues.

**Comment by dsrivastavv** [comment]
> > LGTM! @squirrelsc / @LiliDeng Can you please take a look?
> > @darkwhite29 Please fix any pipeline issues
> 
> You mean all issues from "nox -vRt all"? There are no issues.

They seem to be fixed now. The last PR had one test failing.

File: microsoft/testsuites/osu/osusuite.py [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+        ).is_equal_to(expected_count)
+
+        out_path = node.get_working_path() / "osu-micro-benchmarks-6.2"
+        node.execute(
+            f"mpirun -np {gpu_count} python3 {in_path}/python/run.py"
+            f" --benchmark allgather --min 128 --max 10000000 >> "
+            f"{out_path}/osu_bench.log",  # --buffer pycuda/cupy
+            shell=True,
+            sudo=True,
+        )
```
Comment (LiliDeng) [inline microsoft/testsuites/osu/osusuite.py]: here no error code check, no expected value from osu_bench.log, the case will always pass on this step

Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: Added exit code check. Performance number parsing in osu_bench.log is left as future work, since threshold values need to be defined which involves multi-parties' efforts. Please take a final look for now. Thanks.

File: microsoft/testsuites/osu/osusuite.py [hunk 20]
```diff
@@ -0,0 +1,134 @@
...
+            "bcast",
+            "gather",
+            "reduce_scatter",
+            "reduce",
+            "scatter",
+        ]
+        for test in tests:
+            node.execute(
+                f"mpirun -np {gpu_count} python3 {in_path}/python/run.py"
+                f" --benchmark {test} --min 128 --max 10000000 >> "
```
Comment (LiliDeng) [inline microsoft/testsuites/osu/osusuite.py:84]: when I tried this command with gpu_count = 1, it said.
```
This test requires at least two processes
```
I checked the output of `mpirun -h`, so it should be cpu count.
```
-n/-np {value}                   number of processes
```

**Comment by LiliDeng** [comment]
Test results 
``` 
OSU Bench.run_osu_bench: PASSED   
```
[osu_bench.log](https://github.com/microsoft/lisa/files/10220284/osu_bench.log)

File: microsoft/testsuites/osu/osusuite.py [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+    extracted_path = tar.get_tool_path(use_global=True)
+    if node.shell.exists(path / OSU_BENCH_FILE_NAME):
+        return str(extracted_path) + "/osu-micro-benchmarks-6.2"
+
+    download_path = wget.get(
+        url=OSU_BENCH_LOCATION, filename=str(OSU_BENCH_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(extracted_path))
+    assert isinstance(node.os, Posix)
+    node.os.install_packages("python3.8-dev")
```
Comment (squirrelsc) [inline microsoft/testsuites/osu/osusuite.py]: It's highly dependent on os distro. We have the Python/pip tool already. It's ok in this PR. But it should be considered to use the tools in next refactoring.

**Comment by dsrivastavv** [comment]
@darkwhite29 Can you please resplit the commit into two separate commits: change in infiniband and osu test? You can do it by first doing a soft reset with `git reset --soft HEAD~2` and then readding the files into two separate commits. Please feel free to ping me if you are facing any problem

File: microsoft/testsuites/osu/osusuite.py (hunk 21 above)
Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py]: Will do next time. Thanks!

File: microsoft/testsuites/osu/osusuite.py [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+    node.execute(
+        "curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py",
+        shell=True,
+        sudo=True,
+        expected_exit_code=0,
+    )
+    node.execute("python3.8 get-pip.py", shell=True, sudo=True, expected_exit_code=0)
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py:150]: @LiliDeng Any reason to split them into multiple pip calls?

**Comment by darkwhite29** [comment]
> @darkwhite29 Can you please resplit the commit into two separate commits: change in infiniband and osu test? You can do it by first doing a soft reset with `git reset --soft HEAD~2` and then readding the files into two separate commits. Please feel free to ping me if you are facing any problem

Done.

File: microsoft/testsuites/osu/osusuite.py [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+
+
+def _install_osu_mpi(node: Node) -> None:
+    assert isinstance(node.os, Posix)
+    node.os.install_packages("libibverbs-dev gfortran bison")
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = wget.get_tool_path(use_global=True)
+    extracted_path = tar.get_tool_path(use_global=True)
```
Comment (dsrivastavv) [inline microsoft/testsuites/osu/osusuite.py]: @LiliDeng This will extract folder in a system directory. Was this change intended?

Comment (LiliDeng) [reply microsoft/testsuites/osu/osusuite.py]: I just used previous code and didn't notice this.

Comment (dsrivastavv) [reply microsoft/testsuites/osu/osusuite.py]: No problem. I thought these were intentional. @darkwhite29 Do you have a copy of changes we worked on prior to Lili's changes?

File: microsoft/testsuites/osu/osusuite.py (hunk 22 above)
Comment (LiliDeng) [reply microsoft/testsuites/osu/osusuite.py:150]: @darkwhite29 please use one command to do pip install

Comment (darkwhite29) [reply microsoft/testsuites/osu/osusuite.py:150]: Done.

**Review by dsrivastavv — DISMISSED** [review]
LGTM! @LiliDeng The latest commit contains your changes and the ones I suggested. Can you please take another look?

File: microsoft/testsuites/osu/osusuite.py [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+    path = node.get_working_path()
+    if node.shell.exists(path / OSU_BENCH_FILE_NAME):
+        return str(path) + "/osu-micro-benchmarks-6.2"
+
+    download_path = wget.get(
+        url=OSU_BENCH_LOCATION, filename=str(OSU_BENCH_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(path))
+    assert isinstance(node.os, Posix)
+    node.os.install_packages("python3.8-dev")
```
Comment (LiliDeng) [inline microsoft/testsuites/osu/osusuite.py]: change it into `node.os.install_packages("python3.8-dev python3.8-distutils")`, others are good!

**Review by LiliDeng — APPROVED** [review]
